test/unit/BPool/BPool_Unbind.t.sol

```diff
@@ -40,38 +40,26 @@ contract BPoolUnbind is BPoolBase {
     bPool.unbind(token);
   }
 
-  modifier whenTokenIsBound() {
+  function test_RevertWhen_PoolIsFinalized() external whenCallerIsController {
+    _setRecord(token, IBPool.Record({bound: true, index: 0, denorm: 0}));
```

Comment:
note that it doesn't need ALL `whenTokenCanBeUnbound` expectations, for example, the `totalWeight` in this context is zero, so the method would underflow, but it doesn't matter bc if fails 1st with the finalized check.

the `_setRecord` comes right after the `when token is not bound` branch, so we can assume that after that branch the token will be bound, and that's why i'm allowing myself to add this line..

is a bit implementation-based, since we cannot know from beforehand if a modifier is needed (if there were more reverting checks, perhaps adding a common modifier would save us LoCs, but i feel at least is consistent